### PR TITLE
US156382 - vdiff README

### PR DESCRIPTION
**Pull request by svanherk**
A number of the links here may change as we flesh out https://github.com/BrightspaceUI/testing's documentation, so we should come back here afterwards and check.

### CODE REVIEW

File: vdiff/README.md [hunk 1]
```diff
@@ -1,3 +1,86 @@
 # vdiff Action
 
-TO DO
+This GitHub action runs your repo's [vdiff tests](https://github.com/BrightspaceUI/testing#vdiff-testing). If the tests fail, a draft PR will be opened with the new goldens against the branch/PR that triggered the action. If they pass, any open golden PRs for that branch/PR will be closed for you.
+
+More specifically, this action will:
+* Cleanup abandoned golden PRs
+* Grab your current golden images and set them up for the test run
+* Run your vdiff tests and display the results in a custom commit status (including error messages, a link to the golden PR, etc.)
+  * If failures occur, it will open/update the corresponding golden PR with the new goldens and link to the report stored in S3
```
Comment (dlockhart) [inline vdiff/README.md:9]: The fact that the reports are currently stored in S3 seems irrelevant?

```suggestion
  * If failures occur, it will open/update the corresponding golden PR with the new goldens and link to the report
```

File: vdiff/README.md [hunk 2]
```diff
@@ -1,3 +1,86 @@
...
+  * If the tests pass, it will close the corresponding golden PR if it exists
+
+## Using the Action
+
+Typically this action is triggered from a workflow that runs on your pull requests, ensuring that your new code changes did not cause unexpected visual changes before they are merged.
```
Comment (dlockhart) [inline vdiff/README.md:14]: Nit: I've been trying lately to write docs without "your/you" as it's typically unnecessary. For example, this sentence could be:

```suggestion
Typically this action is triggered from a workflow that runs on pull requests, ensuring that new code changes did not cause unexpected visual changes before they are merged.
```

File: vdiff/README.md [hunk 3]
```diff
@@ -1,3 +1,86 @@
...
+
+`d2l-test-runner` Inputs:
+* `test-chrome`: Set `--chrome` flag to run vdiff tests in Chromium. If no browser flags are set, the default `d2l-test-runner` browser(s) for `vdiff` will be used.
+* `test-config`: Set `--config` flag with `d2l-test-runner` config location.
+* `test-files`: Set the `--files` flag with the pattern of vdiff tests to run.
+* `test-firefox`: Set `--firefox` flag to run vdiff tests in Firefox. If no browser flags are set, the default `d2l-test-runner` browser(s) for `vdiff` will be used.
+* `test-safari`: Set `--safari` flag to run vdiff tests in Webkit. If no browser flags are set, the default `d2l-test-runner` browser(s) for `vdiff` will be used.
+* `test-timeout`: Set the `--timeout` flag with the timeout threshold for vdiff tests (in ms).
+
+See the [`testing` repo's README](https://github.com/BrightspaceUI/testing#running-tests) to learn more about these flags.
```
Comment (dlockhart) [inline vdiff/README.md:58]:
```suggestion
See the [`@brightspace-ui/testing` repo's README](https://github.com/BrightspaceUI/testing#running-tests) to learn more about these flags.
```

File: vdiff/README.md [hunk 4]
```diff
@@ -1,3 +1,86 @@
...
+
+**Notes:**
+* You can also run this action in your release workflow to confirm the `main` branch is in a good state before releasing.  If there's a problem, a PR will be opened against `main` to get the goldens back in the expected state.  This mostly comes down to a time versus risk trade-off - the risk of things getting out of sync may be lower than the time taken to run your vdiff tests every release.
+* You can use the standard `GITHUB_TOKEN` that exists automatically, but will need to [setup the AWS secrets](#setting-up-aws-access-creds).
```
Comment (dlockhart) [inline vdiff/README.md:62]: lol

```suggestion
* You can use the standard `GITHUB_TOKEN` that exists automatically, but will need to [set up the AWS secrets](#setting-up-aws-access-creds).
```

File: vdiff/README.md [hunk 5]
```diff
@@ -1,3 +1,86 @@
...
+
+```
+*       @<your_team> <some_username>
+golden/
+.vdiff.json
+```
+
+## Writing Visual Diff Tests
+
+For more info on setting up and writing vdiff tests, you can checkout the [testing repo's README](https://github.com/BrightspaceUI/testing#vdiff-testing).
```
Comment (dlockhart) [inline vdiff/README.md:80]:
```suggestion
For more info on setting up and writing vdiff tests, refer to the [@brightspace-ui/testing repo documentation](https://github.com/BrightspaceUI/testing#vdiff-testing).
```

File: vdiff/README.md (hunk 2 above)
Comment (svanherk) [reply vdiff/README.md]: Will go through and reword these

**Review by dlockhart — APPROVED** [review]
Looks great!
